proto/workload.proto

```diff
@@ -71,6 +71,15 @@ message Workload {
   // NOTE: this *only* includes Selector based policies. Namespace and global polices
   // are returned out of band.
   repeated string authorization_policies = 16;
+
+  WorkloadStatus status = 17;
+}
+
+enum WorkloadStatus {
+  // Workload is healthy and ready to serve traffic.
+  HEALTHY = 0;
+  // Workload is unhealthy and NOT ready to serve traffic.
+  UNHEALTHY = 1;
```

Comment:
Thinking out loud about the ordering here, should the default Workload Status be UNHEALTHY or HEALTHY? It feels to me that `UNHEALTHY` might be a better default. UNHEALTHY until proven healthy maybe.

Reply:
I mostly picked this to align with Envoy's API - but not sure if that is good.

In practice this should always always be explicitly set. In a hypothetical broken control plane that never set this it does seem like we would prefer 100% healthy than 100% unhealthy, but we should also not have broken control planes...

Reply:
Yea, I think if Envoy is this way, then maybe it's better to keep it in line, but I do think the other way makes more sense to me, with the addition of some panic threshold.

Reply:
The states are not set in stone anyway, we will likely have to extend or rework the states when we get to actually implementing e.g. the [ProxyTerminatingEndpoint KEP](https://github.com/istio/ztunnel/issues/375) in ZTunnel. We can revisit then if needed.